### PR TITLE
feat: improvements to on-prem deploy workflow

### PR DESCRIPTION
**Pull request by paulswartz**
- Run `docker pull` on all servers, so that the image is available immediately to all servers for startup (@PaulJKim )
- support specifying `update-order`, so applications which can tolerate >1 instance running do not need downtime
- include comment in the SSM Run Command about what is being deployed
- improved documentation in the workflow

### CODE REVIEW

File: .github/workflows/deploy-on-prem.yml [hunk 1]
```diff
@@ -84,7 +92,7 @@ jobs:
       - name: Deploy to ${{ inputs.on-prem-cluster }}
         id: deploy
         run: |
-          aws ssm send-command --document-name "${{ inputs.document-name }}" --document-version "\$LATEST" --targets '[{"Key":"resource-groups:Name","Values":["${{ inputs.on-prem-cluster }}"]}]' --parameters '{"ECRRepo":["${{ secrets.docker-repo }}"],"DockerTag":["${{ steps.docker.outputs.tag }}"],"TaskCpu":["${{ inputs.task-cpu }}"], "TaskMemory": ["${{ inputs.task-memory }}"],"TaskPort":["${{ inputs.task-port }}"],"ServiceName":["${{ inputs.app-name }}-${{ inputs.environment }}"],"EnvironmentSecret":["${{ inputs.app-name }}-${{ inputs.environment }}-environment"],"SplunkIndex":["${{ inputs.splunk-index }}"]}' --timeout-seconds 600 --max-concurrency "1" --max-errors "0" --region ${{ inputs.aws-region  }} | tee document-output.json
```
Comment (PaulJKim) [inline .github/workflows/deploy-on-prem.yml:87]: So was the reason why the prod deploy I did yesterday took so long that the host that the manager was trying to deploy to didn't have the image already?

Comment (paulswartz) [reply .github/workflows/deploy-on-prem.yml:87]: I believe so, yes. It takes a while to download the image, and `realtime-signs-*` are set to stop the original image before starting the new one. With this change, all the servers will download and cache the image, so any of them should be able to start the new service immediately after terminating the existing container.

**Review by PaulJKim — APPROVED** [review]
Makes sense to me. Thank you for digging into it!

**Comment by PaulJKim** [comment]
@paulswartz I was trying a branch deploy to dev to test out a change and it looks like deploys for realtime-signs are broken. Appears to be a validation error with the addition of the `--comment` option being passed with the command

https://github.com/mbta/realtime_signs/actions/runs/3439085037/jobs/5736202004

**Comment by paulswartz** [comment]
@PaulJKim can you try using the `ps-onprem-concurrency` branch in the deploy to see if that helps? https://github.com/mbta/workflows/pull/8

**Pull request merged**